src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsTrackingOptInRequestsOasis.java

```diff
@@ -0,0 +1,340 @@
+package com.wikia.webdriver.testcases.adstests;
+
+import com.wikia.webdriver.common.core.annotations.NetworkTrafficDump;
+import com.wikia.webdriver.common.core.url.Page;
+import com.wikia.webdriver.common.dataprovider.TrackingOptInDataProvider;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.TrackingOptInModal;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+public class TestAdsTrackingOptInRequestsOasis extends NewTestTemplate {
+
+    private static final Page ADS_HOME_PAGE = new Page("project43", "Project43_Wikia");
+    private static final Page ADS_HIVI_PAGE = new Page("project43", "SyntheticTests/UAP/HiVi");
+
+
+    @NetworkTrafficDump(useMITM = true)
+    @Test(
+            dataProviderClass = TrackingOptInDataProvider.class,
+            dataProvider = "adsKikimoraDataProvider",
+            groups = "AdsOptInRejectedOasis"
+    )
+    public void adsTrackingRejectedForKikimora(String[] instantGlobals, List<String> urlPatterns) {
+        networkTrafficInterceptor.startIntercepting();
+        TrackingOptInModal.setGeoCookie(driver,"EU", "PL");
```

Comment:
This two lines are used in almost every test. I'd propose to extract it to a method. It'd be much easier to perform any change in one place, than in N places.